### ISSUE TITLE
Initialize Connection Info in Resource Instance

### ISSUE THREAD

**Issue by tk8817** [squat/terraform-provider-vultr#4]
Hey @squat, 

For other cloud providers I noticed that when you perform `provisioner` blocks on a `resource` , you can get away with very little info in the `connection` block. This is due to the `provider` initializing / defaulting certain fields as part of the `resource_instance.go` code.

Current Example Usage for Vultr:
```
  provisioner "remote-exec" {
    inline = ["cd /tmp","ls -la"]
    connection {
      host        = "${self.ipv4_address}"
      type        = "ssh"
      user        = "root"
      password    = "${self.default_password}"
      private_key = "${file("~/.ssh/jcommu.pem")}"
      timeout     = "2m"
    }
  }
```

To get the above working I needed to specify the `host`, `type`, `user`, `password`, and `private_key`. Based on how Vultr provisions, we should be able to let users of this provider call a `provisioner` block without passing any `connection` info. 

Requested Usage for Vultr:
```
  provisioner "remote-exec" {
    inline = ["cd /tmp","ls -la"]
  }
```

How to make this work?
In `resource_instance.go` as part of the `resourceInstanceRead ` function, add the following

```
	// Initialize the connection info
	d.SetConnInfo(map[string]string{
		"type": "ssh",
		"host": --get IPv4 from read--,
                "user": "root",
                "password": --getDefaultPassword from read--,
	})
```

In this case, a user could provide a `private_key` or `user` as an over-ride if the like, but it will not be required.

**Comment by tk8817** [comment]
For reference, you can see a working example here in the digital ocean provider

https://github.com/terraform-providers/terraform-provider-digitalocean/blob/4140328ce2285507715e739fca8ba4f5c918989a/digitalocean/resource_digitalocean_droplet.go

**Comment by squat** [comment]
@tk8817, thanks for filing this issue. Provisioning without specifying connection information is indeed broken. We actually already had most of those connection details specified in the instance resource [0], however, I just tested provisioning again with the following config:
```terraform
  provisioner "remote-exec" {
    inline = ["pwd"]
  }
```
and saw that unless you specify a host, you will get the following error over and over:
```
vultr_instance.example (remote-exec): Connecting to remote host via SSH...
vultr_instance.example (remote-exec):   Host: <nil>
vultr_instance.example (remote-exec):   User: root
vultr_instance.example (remote-exec):   Password: true
vultr_instance.example (remote-exec):   Private key: false
vultr_instance.example (remote-exec):   SSH Agent: true
vultr_instance.example: Still creating... (1m50s elapsed)
```
This is because the connection details are specified before the instance is ready and the public IP is actually known. The fix is trivial and I'll post it shortly.
[0] https://github.com/squat/terraform-provider-vultr/blob/master/vultr/resource_instance.go#L177

**Comment by squat** [comment]
fixed in 0dd428df4a1daf8a5a614e84807bbded8b9a0906. Thanks again!
